src/core/diags/factory_diags.cpp

```diff
@@ -1112,6 +1095,8 @@ Error Diags::ProcessCmd(uint8_t aArgsLength, char *aArgs[])
         ExitNow();
     }
 
+    VerifyOrExit(otPlatDiagModeGet() || StringMatch(aArgs[0], "start"), error = kErrorInvalidState);
```

Comment:
You cold add info here that diag is not enabled.

Reply:
good point, added